addons/popochiu/editor/main_dock/popochiu_row/audio_row/popochiu_audio_row.gd

```diff
@@ -0,0 +1,223 @@
+	
+	if is_instance_valid(audio_cue):
+		_delete_dialog.title = "Remove %s cue" % audio_cue.resource_name
+		_delete_dialog.message = DELETE_AUDIO_CUE_MSG % audio_cue.resource_name
+		_delete_dialog.ask = DELETE_AUDIO_CUE_ASK % audio_cue.audio.resource_path
+		_delete_dialog.on_confirmed = _remove_from_popochiu
+	else:
+		_delete_dialog.title = "Delete %s" % file_name
+		_delete_dialog.message = DELETE_AUDIO_FILE_MSG % path
+		_delete_dialog.on_confirmed = _delete_from_file_system
+	
+	PopochiuEditorHelper.show_delete_confirmation(_delete_dialog)
+
+
+#endregion
+
+#region Public #####################################################################################
+func select() -> void:
+	EditorInterface.select_file(audio_cue.resource_path)
+	EditorInterface.edit_resource(audio_cue)
+	super()
+
+
+#endregion
+
+#region Private ####################################################################################
+func _get_menu_cfg() -> Array:
+	return [
+		{
+			id = AudioOptions.ADD_TO_MUSIC,
+			icon = preload("res://addons/popochiu/icons/music.png"),
+			label = "Add to Music"
+		},
+		{
+			id = AudioOptions.ADD_TO_SFX,
+			icon = preload("res://addons/popochiu/icons/sfx.png"),
+			label = "Add to Sound effects"
+		},
+		{
+			id = AudioOptions.ADD_TO_VOICE,
+			icon = preload("res://addons/popochiu/icons/voice.png"),
+			label = "Add to Voices"
+		},
+		{
+			id = AudioOptions.ADD_TO_UI,
+			icon = preload("res://addons/popochiu/icons/ui.png"),
+			label = "Add to Graphic interface"
+		}
+	] + super()
+
+
+func _menu_item_pressed(id: int) -> void:
+	match id:
+		AudioOptions.ADD_TO_MUSIC:
+			group_selected.emit(PopochiuResources.AudioTypes.MUSIC)
+		AudioOptions.ADD_TO_SFX:
+			group_selected.emit(PopochiuResources.AudioTypes.SOUND_EFFECT)
+		AudioOptions.ADD_TO_VOICE:
+			group_selected.emit(PopochiuResources.AudioTypes.VOICE)
+		AudioOptions.ADD_TO_UI:
+			group_selected.emit(PopochiuResources.AudioTypes.UI)
+		_:
+			super(id)
+
+
+func _play() -> void:
+	if (
+		is_instance_valid(audio_tab.last_played)
+		and audio_tab.last_played.get_instance_id() != get_instance_id()
+	):
+		audio_tab.last_played._stop()
+	
+	if not is_instance_valid(audio_cue):
+		var stream: AudioStream = load(path)
+		stream.loop = false
+		stream_player.stream = stream
+	elif (
+		not stream_player.stream
+		or stream_player.stream.resource_path != audio_cue.audio.resource_path
+	):
+		stream_player.stream = audio_cue.audio
+		stream_player.pitch_scale = audio_cue.get_pitch_scale()
+		stream_player.volume_db = audio_cue.volume
+	
+	if stream_player.is_playing():
+		current_playback_position = stream_player.get_playback_position()
+		stream_player.finished.disconnect(_stop)
+		stream_player.stop()
+		play_btn.icon = play_btn.get_theme_icon("MainPlay", "EditorIcons")
+	else:
+		stream_player.finished.connect(_stop)
+		stream_player.play(current_playback_position)
+		play_btn.icon = play_btn.get_theme_icon("Pause", "EditorIcons")
+		audio_tab.last_played = self
```

Comment:
I see here you're toggling play/pause function on the same button. It's ok but I'm not sure if this can get out of sync with the code at the top of this function, when you test if a stream is playing.

Did you consider having a logic representation of the button status (like a boolean) which rules both the button visual AND the button behavior?

Reply:
I didn't. I remember I wanted to simplify (or improve) this, but then I got distracted in another thing. I'll try what you suggest.

Reply:
Done.